crypto/multi_signature.go

```diff
@@ -0,0 +1,89 @@
+package crypto
+
+import (
+	"reflect"
+
+	"github.com/relab/hotstuff"
+	"golang.org/x/exp/slices"
+)
+
+// Signature is the individual component in MultiSignature
+type Signature interface {
+	Signer() hotstuff.ID
+	ToBytes() []byte
+}
+
+// MultiSignature is a set of (partial) signatures.
+type MultiSignature[T Signature] map[hotstuff.ID]T
```

Comment:
Suggest to rename `MultiSignature` to just `Multi` to avoid the stutter. We almost always use it together with something like this `Multi[*Signature]` and so it is still clear that it is a multi signature.